manifest publish panics on MacOS Monterey

<!--
Thank you for helping to improve Upbound! We use issues for bug reports and
feature requests. Please be sure to search for open issues before raising a new
one.
-->

### What happened?
<!--
Please let us know what behaviour you expected and how Upbound diverged from
that behaviour.

If this bug pertains to an existing epic please cross-reference it below, for
example by writing:

Epic https://github.com/upbound/some-repo/issues/1000
-->

When running `make build publish` in the [provider-helm](https://github.com/crossplane-contrib/provider-helm) repository on MacOS Monterey the build panics with the following error:

```
fatal error: runtime: bsdthread_register error

runtime stack:
runtime.throw(0x14b2fa9, 0x21)
        /usr/local/go/src/runtime/panic.go:605 +0x95 fp=0x7ff7bfefdfc0 sp=0x7ff7bfefdfa0 pc=0x1029c35
runtime.goenvs()
        /usr/local/go/src/runtime/os_darwin.go:108 +0x83 fp=0x7ff7bfefdff0 sp=0x7ff7bfefdfc0 pc=0x10274d3
runtime.schedinit()
        /usr/local/go/src/runtime/proc.go:482 +0xa1 fp=0x7ff7bfefe030 sp=0x7ff7bfefdff0 pc=0x102c5e1
runtime.rt0_go(0x7ff7bfefe060, 0x9, 0x7ff7bfefe060, 0x1000000, 0x9, 0x7ff7bfefe410, 0x7ff7bfefe471, 0x7ff7bfefe476, 0x7ff7bfefe480, 0x7ff7bfefe48c, ...)
        /usr/local/go/src/runtime/asm_amd64.s:175 +0x1eb fp=0x7ff7bfefe038 sp=0x7ff7bfefe030 pc=0x10546db
21:23:07 [FAIL]
make[1]: *** [img.release.manifest.publish.provider-helm] Error 1
```

Related issue: https://github.com/golang/go/issues/49425

It seems to fail when executing the manifest tool. The version used is 0.7.0 which was released 2017. A possible solution would be to upgrade the manifest tool to a later version.

### How can we reproduce it?
<!--
Help us to reproduce your bug as succinctly and precisely as possible. Artifacts
such as example manifests or a script that triggers the issue are highly
appreciated!
-->

* Clone [provider-helm](https://github.com/crossplane-contrib/provider-helm)
* Update `DOCKER_REGISTRY` in Makefile to your individual docker user
* Commit the change so you have a clean git history to enable publishing
*  run `make build publish`

### What environment did it happen in?
<!--
Include at least the relevant versions or commits of Upbound software you were
running when you encountered this bug.
-->
[provider-helm](https://github.com/crossplane-contrib/provider-helm) master
MacOS Monterey 12.2
go version go1.17.7 darwin/amd64

--- a/makelib/image.mk
+++ b/makelib/image.mk
@@ -52,7 +52,7 @@ ifeq ($(origin BUILD_REGISTRY), undefined)
 BUILD_REGISTRY := build-$(shell echo $(HOSTNAME)-$(ROOT_DIR) | shasum -a 256 | cut -c1-8)
 endif
 
-MANIFEST_TOOL_VERSION=v0.7.0
+MANIFEST_TOOL_VERSION=v1.0.3
 MANIFEST_TOOL := $(TOOLS_HOST_DIR)/manifest-tool-$(MANIFEST_TOOL_VERSION)
 
 # In order to reduce built time especially on jenkins, we maintain a cache

